pandas/core/frame.py

```diff
@@ -6588,16 +6588,51 @@ def _gotitem(
         **_shared_doc_kwargs
     )
     @Appender(_shared_docs["aggregate"])
-    def aggregate(self, func, axis=0, *args, **kwargs):
+    def aggregate(self, func=None, axis=0, *args, **kwargs):
+        from pandas.core.groupby.generic import (
+            _is_multi_agg_with_relabel,
+            _maybe_mangle_lambdas,
+            _normalize_keyword_aggregation,
```

Comment:
importing private functions is sub-optimal.  it looks like all of the stuff that actually _uses_ these functions doesn't rely on `self`.  could a public function be made in groupby.generic (or possibly SelectionMixin per @jreback's comment) that consists of basically 6600-6611?

Reply:
yes, i made it in the `base` class

and because of this, I think in the follow up PR, I could do a cleaning on the other one in `core/groupby/generic.py`.